zhusuan/model/base.py

```diff
@@ -283,15 +283,18 @@ def _check_names_exist(self, name_or_names):
 
         :param name_or_names: A string or a list of strings. Names of
             `StochasticTensor` s in the network.
+        :return: The validated name, or a tuple of the validated names.
         """
-        if isinstance(name_or_names, (tuple, list)):
-            names = name_or_names
+        if isinstance(name_or_names, six.string_types):
+            names = (name_or_names,)
         else:
-            names = [name_or_names]
+            name_or_names = tuple(name_or_names)
```

Comment:
The main purpose is to support iterator? Then there should be an added test case.